clients/src/main/java/org/apache/kafka/common/requests/ShareFetchResponse.java

```diff
@@ -79,6 +83,29 @@ public Map<Errors, Integer> errorCounts() {
         return counts;
     }
 
+    public LinkedHashMap<TopicIdPartition, ShareFetchResponseData.PartitionData> responseData(Map<Uuid, String> topicNames) {
+        if (responseData == null) {
+            synchronized (this) {
```

Comment:
Are we going to have ShareFetchResponse object handled in multiple threads?

Reply:
Probably not, but I think this pattern is already used and it does make it safe to initialize the data lazily.

Reply:
Hmmm, we can initialize the data lazily with synchronized as well. But I do see similar pattern in FetchResponse.java, may be it's the way it should be but just wanted to see how response object gets shared between threads in clients.